modules/tensor_mechanics/src/materials/abaqus/AbaqusUMATStress.C

```diff
@@ -44,6 +45,8 @@ AbaqusUMATStress::validParams()
   params.addParam<MooseEnum>("decomposition_method",
                              ComputeFiniteStrain::decompositionType(),
                              "Method to calculate the strain kinematics.");
+  params.addParam<UserObjectName>(
```

Comment:
It would be nice if that object could be obtained automatically

Reply:
Automatically by casting? If so, it may be risky as there could be multiple `StepUserObject`s for whatever reason.

Reply:
Yeah, we discussed that, and I'm of the opinion that this should trigger an error. I don;t like that this creates another pitfall for the users. They add the steps but forget to specify the parameter to use the steps.

Reply:
We certainly didn't discuss the possible cases that appear as we enforce this behavior. I think doing what you suggest is fine, but comes at the expense of limiting the the possibilities of using steps.

Reply:
We could have it both ways by searching for a userobjec if it isn't supplied explicitly. The same pattern is used in the ray tracing module, where the _RayTracingStudy_ is an optional parameter to RayKernels and RayBCs. If that parameter is not supplied the RayTracingStudy object is automatically detected, but it has to be unique.

I did mention that having  multiple `StepUserObject`s should not be permitted. But to quote George Bernard Shaw: "The single biggest problem in communication is the illusion that it has taken place".